### PR TITLE
Build scripts separated to "build" folder

### PR DESCRIPTION
**Pull request by xpl**
That somewhat reduces the chaos in the root directory, as it was previously hard to say what files are related to the build process. Also can now exclude the whole `build` directory (not just individual files) in `.npmignore`. Some build scripts were also refactored a bit.

The `ccxt.browser.js` bundle has been moved to `dist` folder.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -121,7 +121,12 @@ The contents of the repository are structured as follows:
 /CONTRIBUTING.md           # this file
 /LICENSE.txt               # MIT
 /README.md                 # master markdown for GitHub, npmjs.com, npms.io, yarn and others
-/build/                    # a folder for the generated files
+/build/                    # build scripts
+/build/export-exchanges.js # used to create tables of exchanges in the docs during the build
+/build/transpile.js        # the transpilation script
+/build/update-badges.js    # a JS script to update badges in the README and in docs
+/build/vss.js              # reads single-sourced version from package.json and writes it everywhere
+/dist/                     # a folder for the generated browser bundle of CCXT
```
Comment (kroitor) [inline CONTRIBUTING.md]: Have you checked the python build process? There might be a conflict with `dist` being reserved for python `bdist` and `sdist`.

Comment (xpl) [reply CONTRIBUTING.md]: I checked the `npm run build` and `setup.py` scripts, it works. Seems that it Python creates a `dist` folder inside of the `python` folder (`python/dist`), not in the root dir...

**Comment by xpl** [comment]
Hmm, looks like it fails to build on Travis, I'll figure out.

**Comment by kroitor** [comment]
https://travis-ci.org/ccxt/ccxt/builds/500794429?utm_medium=notification&utm_source=email

**Comment by kroitor** [comment]
The problem is this line, i guess: https://travis-ci.org/ccxt/ccxt/builds/500794429#L2249

**Comment by kroitor** [comment]
The CDN stopped working:

- https://github.com/ccxt/ccxt/issues/4773
- https://unpkg.com/ccxt
- https://unpkg.com/ccxt@1.18.315/dist/ccxt.browser.js

**Pull request merged**